Firebase/InstanceID/FIRInstanceIDCheckinService.m

```diff
@@ -179,6 +188,8 @@ - (void)checkinWithExistingCheckin:(FIRInstanceIDCheckinPreferences *)existingCh
 
 - (void)stopFetching {
   [self.session invalidateAndCancel];
+  // The session cannot be reused after invalidation. Dispose it to prevent accident reusing.
+  self.session = nil;
```

Comment:
is it possible that a stopFetching is called and another fetch is requested? this way self.session will never be initiated.

Reply:
As far as I can see from the code, the only cases when it is called is before deallocating `FIRInstanceIDCheckinService` or `FIRInstanceIDTokenManager`. If it is true and I haven't missed anything, then a next request unlikely should be sent using the same `FIRInstanceIDCheckinService` instance.

But I agree, creating a new instance sounds like a better approach.

@chliangGoogle @ryanwilson Would you like me to update it, or keep it as is so we don't introduce too much new logic without knowing the core reason of the issue?

Reply:
Let's take this out for now to avoid adding new logic.

Reply:
@chliangGoogle Well, this is the core of the fix. If we don't set it to `nil` or to a new valid session, the service will keep crashing.

Reply:
yeah confirm it is safe to do that. Feel free to submit the PR once travis is green.